src/cascadia/TerminalControl/ControlInteractivity.cpp

```diff
@@ -0,0 +1,560 @@
+        {
+            const auto clickCount = _numberOfClicks(mouseCursorPosition, timestamp);
+            // This formula enables the number of clicks to cycle properly
+            // between single-, double-, and triple-click. To increase the
+            // number of acceptable click states, simply increment
+            // MAX_CLICK_COUNT and add another if-statement
+            const unsigned int MAX_CLICK_COUNT = 3;
+            const auto multiClickMapper = clickCount > MAX_CLICK_COUNT ? ((clickCount + MAX_CLICK_COUNT - 1) % MAX_CLICK_COUNT) + 1 : clickCount;
+
+            // Capture the position of the first click when no selection is active
+            if (multiClickMapper == 1)
+            {
+                _singleClickTouchdownPos = mouseCursorPosition;
+                _singleClickTouchdownTerminalPos = terminalPosition;
+
+                if (!_core->HasSelection())
+                {
+                    _lastMouseClickPosNoSelection = mouseCursorPosition;
+                }
+            }
+            const bool isOnOriginalPosition = _lastMouseClickPosNoSelection == mouseCursorPosition;
+
+            _core->LeftClickOnTerminal(terminalPosition,
+                                       multiClickMapper,
+                                       altEnabled,
+                                       shiftEnabled,
+                                       isOnOriginalPosition,
+                                       _selectionNeedsToBeCopied);
+
+            if (_core->HasSelection())
+            {
+                // GH#9787: if selection is active we don't want to track the touchdown position
+                // so that dragging the mouse will extend the selection rather than starting the new one
+                _singleClickTouchdownPos = std::nullopt;
+            }
+        }
+        else if (buttonState.isRightButtonDown)
+        {
+            // CopyOnSelect right click always pastes
+            if (_core->CopyOnSelect() || !_core->HasSelection())
+            {
+                PasteTextFromClipboard();
+            }
+            else
+            {
+                CopySelectionToClipboard(shiftEnabled, nullptr);
+            }
+        }
+    }
+
+    void ControlInteractivity::Touched(const winrt::Windows::Foundation::Point contactPoint)
+    {
+        _touchAnchor = contactPoint;
+    }
+
+    void ControlInteractivity::PointerMoved(const winrt::Windows::Foundation::Point mouseCursorPosition,
+                                            TerminalInput::MouseButtonState buttonState,
+                                            const unsigned int pointerUpdateKind,
+                                            const ::Microsoft::Terminal::Core::ControlKeyStates modifiers,
+                                            const bool focused,
+                                            const til::point terminalPosition)
+    {
+        // Short-circuit isReadOnly check to avoid warning dialog

```

Comment:
We don't need the `FocusFollowMouse` stuff here?

Reply:
that's still up in `TermControl`

Reply:
editorialized: FFM should probably have stopped at Pane anyway.